### PR TITLE
Add AWS and Azure quickstarts

### PR DESCRIPTION
**Pull request by endingwithali**
First two quick starts for moving quick with azure and AWS are ready to publish

### CODE REVIEW

File: docs/quickstart/AWS.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+
+Press `enter` to select `Develop in Linux` on AWS template. This will return the following: 
+
+<img src="../images/quickstart/aws/aws10.png">
+
+
+To get started using the AWS template, install the AWS CLI and put the credentials in this account. Follow these instructions to install the AWS CLI and these instructions to add your credentials.
+
+For this example, Coder was run as a system level process, which creates the system user `coder` for handling processes. The coder user will require access to the AWS credentials to initialize the template.
```
[the remark this answers is not in the document]
Comment (bpmct) [reply docs/quickstart/AWS.md:153]: The current verbiage is fine, but I think these suggestions clarify things a bit. Main point being the AWS credentials are necessary for all workspace provisioning (you can't remove them after you initialize the template)

File: docs/quickstart/AWS.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+```
+coder templates create
+```
+
+
+Congrats! You can now navigate to your Coder dashboard and use this Linux on AWS template to create a new workspace!
+
+
+## Debugging help
```
Comment (bpmct) [inline docs/quickstart/AWS.md:177]: Suggest changing it to troubleshooting to be consistent with other sections. 

```suggestion
## Troubleshooting
```

File: docs/quickstart/Azure.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+To get started running Coder on Azure, you will need an Azure account with the capability to create new Virtual Machines. 
+
+## Create An Azure VM
+
+From the Azure Portal, navigate to the Virtual Machines Dashboard. Click Create, and select creating a new Azure Virtual machine .
+
+<img src="../images/quickstart/azure/azure1.jpg">
+
+
+This will bring you to the `Create a virtual machine page`. Select the subscription group of your choice, or create one if necessary.
```
Comment (bpmct) [inline docs/quickstart/Azure.md:11]:
```suggestion
This will bring you to the `Create a virtual machine` page. Select the subscription group of your choice, or create one if necessary.
```

Comment (bpmct) [reply docs/quickstart/Azure.md:11]: assuming this was a typo

File: docs/quickstart/AWS.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+If you’re not sure if Coder is running correctly as a system process, use the command `systemctl status coder` to show the status of the coder process.
+
+
+To stop running coder as a system process, run the following command: `systemctl stop coder`.
```
Comment (bpmct) [inline docs/quickstart/AWS.md:182]:
```suggestion
To stop running Coder as a system process, run the following command: `systemctl stop coder`. 
```

**Review by bpmct — CHANGES_REQUESTED** [review]
Super excited to see these coming together. I noticed some minor typos as well as Markdown  syntax errors. For the content that is repeated for AWS and Azure, I only left a comment on AWS.

File: docs/manifest.json [hunk 5]
```diff
@@ -64,6 +64,16 @@
           "description": "Setup Coder with Docker",
           "icon_path": "./images/icons/docker.svg",
           "path": "./quickstart/docker.md"
+        },
+        {
+          "title": "AWS",
+          "description": "Setup Coder with AWS",
+          "path": "./quickstart/AWS.md"
```
[the remark this answers is not in the document]
Comment (endingwithali) [reply docs/manifest.json:71]: why would i lowercase AWS lol ill change the file name but not the title

File: docs/quickstart/azure.md [hunk 6]
```diff
@@ -95,32 +86,27 @@ Press `enter` to select `Develop in Linux on Azure` template. This will return t
...
-
-
-To stop running coder as a system process, run the following command: `systemctl stop coder`. 
-
-When you run ` journalctl -u coder.service -b`, there’s a chance that it may not show any logs, preventing you from seeing the url for the running coder instance. Use the command `systemctl stop coder` to stop the coder system process, and run the command to start Coder as a system process again. You can also use the command `systemctl restart coder` to quickly restart the process. 
+If you’re not sure if Coder is running correctly as a system service, use the command `systemctl status coder` to show the status of the Coder process.
 
+To stop running Coder as a system process, run the following command: `systemctl stop coder`. 
 
+When you run ` journalctl -u coder.service -b`, there’s a chance that it may not show any logs, preventing you from seeing the url for the running Coder instance. Use the command `systemctl stop coder` to stop the Coder system process, and run the command to start Coder as a system process again. You can also use the command `systemctl restart coder` to quickly restart the process.
```
Comment (ammario) [inline docs/quickstart/azure.md:112]: Why not suggest they use `journalctl --grep` instead? I see your comment in Google Docs: "this is for just seeing if its running?", which I don't understand. This paragraph assumes the service is already running since you're recommending they stop it.

File: docs/quickstart/azure.md [hunk 7]
```diff
@@ -95,32 +86,27 @@ Press `enter` to select `Develop in Linux on Azure` template. This will return t
...
-
-If you’re not sure if Coder is running correctly as a system service, use the command `systemctl status coder` to show the status of the coder process.
-
-
-To stop running coder as a system process, run the following command: `systemctl stop coder`. 
-
-When you run ` journalctl -u coder.service -b`, there’s a chance that it may not show any logs, preventing you from seeing the url for the running coder instance. Use the command `systemctl stop coder` to stop the coder system process, and run the command to start Coder as a system process again. You can also use the command `systemctl restart coder` to quickly restart the process. 
+If you’re not sure if Coder is running correctly as a system service, use the command `systemctl status coder` to show the status of the Coder process.
 
+To stop running Coder as a system process, run the following command: `systemctl stop coder`.
```
Comment (ammario) [inline docs/quickstart/azure.md:110]: We should be consistent about calling the service `system service` instead of `system process`.

**Review by ammario — CHANGES_REQUESTED** [review]

**Comment by ammario** [comment]
@endingwithali I renamed the PR to `Add AWS and Azure quickstarts` instead of `Merge AWS and Azure quickstarts` to be consistent with our other PRs and commits. Beginning a commit with `Merge`, `Rebase` or some git operation is a bad practice since that information is already implied in the Git object.

**Comment by ammario** [comment]
Also, please ensure that this lays out correctly in coder.com before merging.

File: docs/quickstart/aws.md [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+Copy the URL and run the following command to create the first user, either on your local machine or in the AWS EC2 instance terminal. 
+
+```sh
+coder login <url***.try.coder.app>
+```
+
+Fill out the prompts - be sure to save use email and password - these are your admin username and password.
+
+You can now access Coder on your local machine with the relevant  `***.try.coder.app` URL and logging in with the username and password.
+
```
Comment (ammario) [inline docs/quickstart/aws.md:122]: This section is repeated verbatim above.

File: docs/quickstart/aws.md [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+## Creating and Uploading Your First Template
+
+First, run `coder template init` to create your first template. You’ll be given a list of possible templates to use. This tutorial will show you how to set up your Coder instance to create a linux based machine on AWS.
```
Comment (ammario) [inline docs/quickstart/aws.md:125]:
```suggestion
Run `coder template init` to create your first template. You’ll be given a list of possible templates. This tutorial will show you how to set up your Coder instance to create Linux based machines on AWS. 
```

File: docs/quickstart/aws.md [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+
+First, run `coder template init` to create your first template. You’ll be given a list of possible templates to use. This tutorial will show you how to set up your Coder instance to create a linux based machine on AWS. 
+
+<img src="../images/quickstart/aws/aws9.png">
+
+Press `enter` to select `Develop in Linux` on AWS template. This will return the following: 
+
+<img src="../images/quickstart/aws/aws10.png">
+
+To get started using the AWS template, install the AWS CLI and put the credentials in this account. Follow these instructions to install the AWS CLI and add your credentials.
```
Comment (ammario) [inline docs/quickstart/aws.md:133]:
```suggestion
Now, we must install the AWS CLI and authorize the template.
```

File: docs/quickstart/aws.md [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+
+Since we're running Coder as a system level process, a system user `coder` is created in order to run Coder services. The `coder` user will require access to the AWS credentials to initialize the template and provision workspaces.
```
Comment (ammario) [inline docs/quickstart/aws.md:135]:
```
Since we're running Coder as a system level process, a system user `coder` is created in order to run Coder services.
```
This line doesn't make sense to me because running Coder as a systemd service doesn't require running as an unprivileged user, that was a decision we made.

Comment (ammario) [inline docs/quickstart/aws.md:135]:
```suggestion
The Coder service runs under the `coder` user, so we must authenticate `coder` with AWS before it can initialize the template and provision workspaces.
```

File: docs/quickstart/aws.md [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+```
+
+Then run the following commands to copy the credentials and give the `coder` user access to them:
+
+```sh
+sudo cp ~/.aws/credentials /home/coder/.aws/credentials 
+sudo chown coder:coder /home/coder/.aws/credentials
+```
+
+Navigate to the `/aws-linux` folder where you created your template and run the following command to put the template on your Coder instance.
```
Comment (ammario) [inline docs/quickstart/aws.md:150]:
```suggestion
Navigate to the `./aws-linux` folder where you created your template and run the following command to put the template on your Coder instance. 
```

`/` implies root

**Review by ammario — CHANGES_REQUESTED** [review]
I didn't thoroughly review the Azure Quickstart yet

File: docs/manifest.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (bpmct) [reply docs/manifest.json]: Yeah I meant for the `path` (filename) not the `title`

File: docs/quickstart/aws.md [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+
+```sh
+chmod 400 [mykey].pem
+```
+
+This adds the required permissions for ssh-ing into an EC2 instance.
+
+Run the following command in terminal, where `mykey` is the security key file, `username` is the username found above for the relevant EC2 operating system image, and the `ip-address` is the IPv4 address for the server: 
+
+```
```
Comment (bpmct) [inline docs/quickstart/aws.md:58]:
```suggestion
```sh
```

**Comment by endingwithali** [comment]
> Also, please ensure that this lays out correctly in coder.com before merging.

checked by building and running docs locally

File: docs/quickstart/aws.md (hunk 11 above)
Comment (endingwithali) [reply docs/quickstart/aws.md:135]: we dont actually end up authenticating the coder user but instead just copying and giving it access to the aws credentials.  solution is a blend of the two sentences above

Comment (endingwithali) [reply docs/quickstart/aws.md:135]: never said it was unpriv however it does run as a different user?

File: docs/quickstart/azure.md (hunk 6 above)
Comment (endingwithali) [reply docs/quickstart/azure.md:112]: specific command was at the instruction of ben so that was why i included it as the choice instruction

File: docs/quickstart/aws.md (hunk 11 above)
Comment (ammario) [reply docs/quickstart/aws.md]: I'm using "unpriv" to refer to the non-root user. I think my confused interpretation stems from the use of the "system level process" language, when it's the systemd unit creating the user and not the `coder server`. I believe you've rephrased a lot of the doc so I will re-review when requested.

Comment (ammario) [reply docs/quickstart/aws.md]: I'm confused by what you mean. The copying of the credentials into the `coder` user is authenticating it in my eyes?

Comment (endingwithali) [reply docs/quickstart/aws.md]: when i think authentication i think using the established pattern to authenticate a specific user - ie using aws config command when on the coder user to give it credentials - in this case we are copying over the credentials from authenticating the ec2-user over to the coder user

Comment (ammario) [reply docs/quickstart/aws.md]: Okay, I can see it that way too. My goal was to compress the words as much as possible.

Comment (bpmct) [reply docs/quickstart/aws.md]: How about a compromise? 

> Since the Coder service runs under the `coder` user, we need to move the AWS credentials there.

`aws configure` doesn't really do any authentication either, it's just a utility to generate a text file. The real authentication happens when `coder templates <create/push>` or `coder workspace <create/update>` occurs.

Comment (endingwithali) [reply docs/quickstart/aws.md]: regarding word compression my dev rel prerogative is to be as verbose as possible to any possible avoid confusion.

File: docs/quickstart/aws.md [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+
+```sh
+ssh -i [mykey].pem username@ip-address
+```
+
+Congrats you’ve SSH’d into the server.
+
+## Install Coder
+
+For this instance, we will run Coder as a service command, however you can run Coder a multitude of different ways. You can learn more about those [here](https://coder.com/docs/coder-oss/latest/install).
```
Comment (bpmct) [inline docs/quickstart/aws.md:66]: I missed this in my first pass

```suggestion
For this instance, we will run Coder as a system service, however you can run Coder a multitude of different ways. You can learn more about those [here](https://coder.com/docs/coder-oss/latest/install).
```

File: docs/quickstart/azure.md [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+
+The set up for the image is complete at this stage. Click `Review and Create` - review the information and click `Create`. A popup will appear asking you to download the key pair for the server. Click `Download private key and create resource` and place it into a folder of your choice on your local system. 
+
+<img src="../images/quickstart/azure/azure5.png">
+
+Click `Return to create a virtual machine`. Your VM will start up! 
+
+<img src="../images/quickstart/azure/azure6.png">
+
+Click `Go to resource` in the Virtual machine and copy the public IP address. You will need it to SSH into the virtual machine via your local machine.
```
Comment (bpmct) [inline docs/quickstart/azure.md:30]:
```suggestion
Click `Go to resource` in the virtual machine and copy the public IP address. You will need it to SSH into the virtual machine via your local machine.
```

File: docs/quickstart/azure.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+
+Congrats! You can now navigate to your Coder dashboard and use this Linux on Azure template to create a new workspace!
+
+## Troubleshooting
+
+If you’re not sure if Coder is running correctly as a system service, use the command `systemctl status coder` to show the status of the Coder process.
+
+To stop running Coder as a system service, run the following command: `systemctl stop coder`. 
+
+When you run ` journalctl -u coder.service -b`, there’s a chance that it may not show any logs, preventing you from seeing the url for the running Coder instance. Use the command `systemctl stop coder` to stop the Coder system service, and run the command to start Coder as a system service again. You can also use the command `systemctl restart coder` to quickly restart the service.
```
Comment (bpmct) [inline docs/quickstart/azure.md:113]: Should this be all in one paragraph?

**Review by bpmct — APPROVED** [review]
Did another review and it LGTM. I found some tiny things and another point of feedback is you mention both "system level service" and "system service." I prefer the latter but mostly would prefer it to be consistent.

File: docs/quickstart/aws.md (hunk 11 above)
Comment (ammario) [reply docs/quickstart/aws.md]: There's a middle ground.. After we replace "system level process" with "system service", this paragraph becomes:

```
Since we're running Coder as a system service, a system user `coder` is created in order to run Coder services. The `coder` user will require access to the AWS credentials to initialize the template and provision workspaces.
```

"Since we're running Coder as a system service, a system user `coder` is created in order to run Coder services." I think would need rework.

File: docs/quickstart/azure.md (hunk 16 above)
Comment (endingwithali) [reply docs/quickstart/azure.md:113]: it can be?

File: docs/quickstart/aws.md [hunk 17]
```diff
@@ -0,0 +1,152 @@
+## Requirements 
+
+This quickstart assumes you have access to admin level access to AWS.
```
Comment (ammario) [inline docs/quickstart/aws.md:3]: "access to admin level access to AWS" 🤔

File: docs/quickstart/aws.md [hunk 18]
```diff
@@ -0,0 +1,152 @@
...
+
+<img src="../images/quickstart/aws/aws2.png">
+
+Create a new `Inbound Rule` that allows for SSH from your computer’s IP address.
+
+You’ve now created a security group that will be used by your EC2 instance.
+
+## Setting Up Your EC2 instance
+
+On the EC2 dashboard, click `Instances`. This will take you to all the EC2 instances you have created. Click `Launch New Instance`. Name the EC2 instance following the naming convention of your choice.
```
Comment (ammario) [inline docs/quickstart/aws.md:23]: We should probably mention that we're creating the instance for the coder server, not the workspaces. I'm thinking about the naive user that clicks to this Quickstart from the website.

File: docs/quickstart/aws.md [hunk 19]
```diff
@@ -0,0 +1,152 @@
...
+
+<img src="../images/quickstart/aws/aws3.png">
+
+For this tutorial, we are going to launch this as the base Ubuntu server. 
+
+For the `Create key pair`, we are using ED25519 and .pem as we will SSH into the instance later in the tutorial.
```
Comment (ammario) [inline docs/quickstart/aws.md:29]:
```suggestion
For the `Create key pair`, we are using ED25519 and `.pem` as we will SSH into the instance later in the tutorial.
```

File: docs/quickstart/aws.md [hunk 20]
```diff
@@ -0,0 +1,152 @@
...
+<img src="../images/quickstart/aws/aws4.png">
+
+Next, under `Network Settings`, change your Firewall security group to Select existing security group and from the resulting dropdown, select the security group you created in the previous section.
```
Comment (ammario) [inline docs/quickstart/aws.md:32]:
```suggestion
Next, under `Network Settings`, change your Firewall security group to select existing security group and from the resulting dropdown, select the security group you created in the previous section.
```

File: docs/quickstart/aws.md [hunk 21]
```diff
@@ -0,0 +1,152 @@
...
+
+<img src="../images/quickstart/aws/aws5.png">
+
+It’ll take a few minutes for it to show up in your existing instances, so take a break as it starts up.
+
+## SSHing into the EC2 instance
+
+If you’ve launched a new EC2 instance following the previous steps of this tutorial, find the username for the EC2 instance [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/connection-prereqs.html). The version launched in the previous steps was a version of the Amazon Linux so the username is `ubuntu`.
+
+You will also need the IP address of the server. Click on the server in the `Instances` dashboard, and copy the IPv4 address
```
Comment (ammario) [inline docs/quickstart/aws.md:44]:
```suggestion
You will also need the IP address of the server. Click on the server in the `Instances` dashboard, and copy the IPv4 address.
```

File: docs/quickstart/aws.md [hunk 22]
```diff
@@ -0,0 +1,152 @@
...
+
+<img src="../images/quickstart/aws/aws6.png">
+
+Now that we’ve gathered all the information you will need to ssh into your EC2 instance, on a terminal on your local system, navigate to the `.pem` file downloaded when you created the EC2 instance. Run the following command:
```
Comment (ammario) [inline docs/quickstart/aws.md:48]:
```suggestion
Now that we’ve gathered all the information you will need to SSH into your EC2 instance, on a terminal on your local system, navigate to the `.pem` file downloaded when you created the EC2 instance. Run the following command:
```

File: docs/quickstart/aws.md [hunk 23]
```diff
@@ -0,0 +1,152 @@
...
+
+```sh
+chmod 400 [mykey].pem
+```
+
+This adds the required permissions for ssh-ing into an EC2 instance.
```
Comment (ammario) [inline docs/quickstart/aws.md:54]:
```suggestion
This adds the required permissions for SSH-ing into an EC2 instance.
```

File: docs/quickstart/aws.md [hunk 24]
```diff
@@ -0,0 +1,152 @@
...
+
+Congrats! You can now navigate to your Coder dashboard and use this Linux on AWS template to create a new workspace!
+
+## Troubleshooting
+
+If you’re not sure if Coder is running correctly as a system service, use the command `systemctl status coder` to show the status of the Coder process.
+
+To stop running Coder as a system service, run the following command: `systemctl stop coder`. 
+
+When you run ` journalctl -u coder.service -b`, there’s a chance that it may not show any logs, preventing you from seeing the url for the running Coder instance. Use the command `systemctl stop coder` to stop the Coder system service, and run the command to start Coder as a system service again. You can also use the command `systemctl restart coder` to quickly restart the service.
```
Comment (ammario) [inline docs/quickstart/aws.md:152]: Should use journalctl grep

**Review by ammario — CHANGES_REQUESTED** [review]

File: docs/quickstart/aws.md (hunk 17 above)
Comment (endingwithali) [reply docs/quickstart/aws.md:3]: i wrote this without figuring out the min perms to get coder working. hence requiring admin access

Comment (ammario) [reply docs/quickstart/aws.md:3]: Yeah I mean the word `access` is redundant. I think you're referring to the `AdministratorAccess` policy?

**Review by bpmct — APPROVED** [review]

File: docs/quickstart/aws.md [hunk 25]
```diff
@@ -0,0 +1,152 @@
+## Requirements 
+
+This quickstart assumes you have access to `AdministratorAccess` policy to AWS.
```
Comment (ammario) [inline docs/quickstart/aws.md:3]:
```suggestion
This quickstart assumes you are assigned the `AdministratorAccess` policy to AWS. 
```

I think this reads better without the double "access".

File: docs/quickstart/aws.md [hunk 26]
```diff
@@ -0,0 +1,152 @@
...
+
+Congrats! You can now navigate to your Coder dashboard and use this Linux on AWS template to create a new workspace!
+
+## Troubleshooting
+
+If you’re not sure if Coder is running correctly as a system service, use the command `systemctl status coder` to show the status of the Coder process.
+
+To stop running Coder as a system service, run the following command: `systemctl stop coder`. 
+
+When you run ` journalctl -u coder.service -b`, there’s a chance that it may not show any logs, preventing you from seeing the url for the running Coder instance. Use the command `systemctl stop coder` to stop the Coder system service, and run the command to start Coder as a system service again. You can also use the command `systemctl restart coder` to quickly restart the service.
```
Comment (ammario) [inline docs/quickstart/aws.md:152]: Where did you end up with the journalctl issue? If you really want to tell them to restart, then why not just instruct `systemctl restart coder` and remove the sentence about `systemctl stop coder`?

File: docs/quickstart/azure.md [hunk 27]
```diff
@@ -0,0 +1,113 @@
...
```
Comment (ammario) [inline docs/quickstart/azure.md:113]: <same as above>

How about we move Troubleshooting to another PR so we can merge the meat of this one?

File: docs/quickstart/aws.md [hunk 28]
```diff
@@ -1,6 +1,6 @@
 ## Requirements 
 
-This quickstart assumes you have access to `AdministratorAccess` policy to AWS. 
+This quickstart assumes you have the ability to assign  `AdministratorAccess` policy to AWS.
```
Comment (ammario) [inline docs/quickstart/aws.md:3]: Sorry for all the back and forth.. this should be "you are assigned `AdministratorAccess` right? Not necessarily the ability to assign the role...

**Review by ammario — CHANGES_REQUESTED** [review]

**Review by ammario — APPROVED** [review]

**Comment by ammario** [comment]
🕺🏽🕺🏽🕺🏽🕺🏽🕺🏽
